src/types/definitions.d.ts

```diff
@@ -1,2 +1,8 @@
+interface Window {
+  web3: {
+    currentProvider: any
```

Comment:
there's provider type in web3.js: https://github.com/ethereum/web3.js/blob/1.x/packages/web3-core/types/index.d.ts#L436

Reply:
also web3 could be undefined too

Reply:
I added several types to `getWeb3.ts` file, please give it a look